### PR TITLE
fix/deps: min versions of electron-chromedriver...

### LINKED ISSUE #23573: Apple macOS 12 ( arm64 ) :: ./script/build : error : script is requesting a distribution of electron/mksnapshot (v9.0.2) that does not exist
### Description

The build script for Atom, after cloning the repository, is attempting to install a version of Electron (v9.0.2) that does not have the files that the script is calling for to build the application.
I took the liberty of checking if it was there, and it is not. I also noticed in other release tags, and in the discussion that later versions do have the necessary files, and even files that are optimized for the operating system (apparently contributors are working from version 11 [ contributor "icecream17", #23500 ]).
In any case, the build script is calling an earlier version of Electron for a file that does not exist therein, so Atom does not build on my machine.
I've even attempted to install the latest version of Electron ( v17.0.0 ), onto my service for npm, which installed fine on npm without any problems, but this does not bypass the issue at hand. It seems as though the build script for Atom is trying create its own runtime environment to set itself apart from whatever I do on npm.

### Steps to Reproduce

1. `git clone https://github.com/atom/atom.git`
2. `cd atom`
3. `./script/build`

**Behavior:**

```
Node:	v17.4.0
Python:	v2.7.18
Installing script dependencies
npm ERR! code 1
npm ERR! path /Users/alines7777/Git/atom/script/node_modules/electron-mksnapshot
npm ERR! command failed
npm ERR! command sh -c node ./download-mksnapshot.js
npm ERR! WARNING: mksnapshot does not run on arm64. Download 
npm ERR!   https://github.com/electron/electron/releases/download/v9.0.2/mksnapshot-v9.0.2-darwin-arm64-x64.zip
npm ERR!   on a x64 darwin OS to generate undefined snapshots.

```

**Reproduces how often:**

Everytime

### Versions

atom and apm have not been built.

output of `uname -a`:

`Darwin [REDACTED].local 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101 arm64
`

Thank you.

### PR DESCRIPTION
**Pull request by icecream17**
... and electron-snapshot to the current electron version (v11)

### Identify the Bug

fixes https://github.com/atom/atom/issues/23573

### Description of the Change

Updates electron-chromedriver and electron-snapshot to the current version (v11) and also adds code in `script/lib/check-chromedriver-version` to make sure that the versions always match.

This fixes the issue because https://github.com/electron/mksnapshot/releases/tag/v11.0.1 supports arm64 macOS

### Alternate Designs

Oh wait, semver has comparison functions like `lt` and `gte`, see my suggested changes.

### Possible Drawbacks

Oops, I ran `npm install npm@6` but forgot to put `-g`, so I guess the npm dependency is also updated (`6.14.8` --> `^6.14.16`)

### Verification Process

tests pass!

### CODE REVIEW

File: script/lib/check-chromedriver-version.js [hunk 1]
```diff
@@ -26,15 +32,15 @@ module.exports = function() {
     );
   }
 
-  if (!semver.satisfies(chromedriverActualVer, '>=9.0.0')) {
+  if (!semver.satisfies(chromedriverActualVer, `>=${majorElectronVersion}`)) {
```
Comment (icecream17) [inline script/lib/check-chromedriver-version.js:35]:
```suggestion
  if (semver.lt(chromedriverActualVer, majorElectronVersion)) {
```

**Review by icecream17 — COMMENTED** [review]
Alternative design using `semver.lt` (less than)

**Comment by darangi** [comment]
Thanks for the contribution 🙇🏾‍♂️   @icecream17

**Comment by alines7777** [comment]
Thank you very much.

I ran into an issue here as follows with the build script ( presumably as it was attempting to install `fs-admin` via `npm` );

```
   4958 verbose node v17.4.0
   4959 verbose npm  v8.4.1
   4960 error code 1
   4961 error path /Users/alines7777/Git/atom/script/node_modules/fs-admin
   4962 error command failed
   4963 error command sh -c prebuild-install || node-gyp rebuild
 * 4964 error CXX(target) Release/obj.target/fs_admin/src/main.o
 * 4964 error   CXX(target) Release/obj.target/fs_admin/src/fs-admin-darwin.o
   4965 error In file included from ../src/main.cc:1:
   4965 error In file included from ../../nan/nan.h:2884:
 * 4965 error ../../nan/nan_typedarray_contents.h:34:43: error: no member named 'GetContents' in 'v8::ArrayBuffer'
 * 4965 error       data   = static_cast<char*>(buffer->GetContents().Data()) + byte_offset;
 * 4965 error                                   ~~~~~~~~^
   4965 error 1 error generated.
   4965 error make: *** [Release/obj.target/fs_admin/src/main.o] Error 1
   4965 error gyp ERR! build error 
   4965 error gyp ERR! stack Error: `make` failed with exit code: 2
   4965 error gyp ERR! stack     at ChildProcess.onExit (/Users/alines7777/.nvm/versions/node/v17.4.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
   4965 error gyp ERR! stack     at ChildProcess.emit (node:events:520:28)
   4965 error gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
   4965 error gyp ERR! System Darwin 21.3.0
   4965 error gyp ERR! command "/Users/alines7777/.nvm/versions/node/v17.4.0/bin/node" "/Users/alines7777/.nvm/versions/node/v17.4.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
   4965 error gyp ERR! cwd /Users/alines7777/Git/atom/script/node_modules/fs-admin
   4965 error gyp ERR! node -v v17.4.0
   4965 error gyp ERR! node-gyp -v v8.4.1
   4965 error gyp ERR! not ok
   4966 verbose exit 1
   4967 timing npm Completed in 88161ms
   4968 verbose unfinished npm timer reify 1644303141494
   4969 verbose unfinished npm timer reify:build 1644303225111
   4970 verbose unfinished npm timer build 1644303225112
   4971 verbose unfinished npm timer build:deps 1644303225112
   4972 verbose unfinished npm timer build:run:install 1644303225181
   4973 verbose unfinished npm timer build:run:install:node_modules/electron-chromedriver 1644303225181
   4974 verbose unfinished npm timer build:run:install:node_modules/electron-mksnapshot 1644303225186
   4975 verbose unfinished npm timer build:run:install:node_modules/fs-admin 1644303225192
   4976 verbose unfinished npm timer build:run:install:node_modules/leveldown 1644303225197
   4977 verbose code 1
```

It looks like a shared library component ( of the C/C++ variety from the looks of it ) is failing to  compile for the installation of `fs-admin`, as it seems to be invoking an element [ `GetContents().Data()` ] that hasn't been declared for the class structure [ `buffer` ] in line 34 of `contents.h`.
Frankly, I'm also unsure as to why member function calls [ a property only of classes in C++, rather than data structures in C ] would even be present in a C-type header file. I think maybe someone should contact the developer of `fs-admin` about this.

**Comment by icecream17** [comment]
The getContents error seems related to this deprecation in oniguruma: https://github.com/atom/node-oniguruma/pull/114, but it doesn't look like the pr directly translates

**Pull request merged**